src/cvode/cvode_bbdpre.c

```diff
@@ -439,83 +447,79 @@ int CVBBDPrecGetNumGfnEvals(void *cvode_mem,
   bbd_data is a pointer to the preconditioner data set by
            CVBBDPrecInit
 
-  Return value:
-  The value returned by this CVBBDPrecSetup function is the int
-    0  if successful,
-    1  for a recoverable error (step will be retried).
+  Return value: SUNLsStatus
   -----------------------------------------------------------------*/
-static int CVBBDPrecSetup(realtype t, N_Vector y, N_Vector fy,
-                          booleantype jok, booleantype *jcurPtr,
-                          realtype gamma, void *bbd_data)
+static SUNLsStatus CVBBDPrecSetup(realtype t, N_Vector y, N_Vector fy,
+                                  booleantype jok, booleantype *jcurPtr,
+                                  realtype gamma, void *bbd_data)
 {
   CVBBDPrecData pdata;
   CVodeMem cv_mem;
+  SUNLsStatus ls_status;
   int retval;
 
   pdata = (CVBBDPrecData) bbd_data;
   cv_mem = (CVodeMem) pdata->cvode_mem;
+  
+  SUNAssignSUNCTX(CV_SUNCTX);
 
   /* If jok = SUNTRUE, use saved copy of J */
   if (jok) {
     *jcurPtr = SUNFALSE;
     retval = SUNMatCopy(pdata->savedJ, pdata->savedP);
-    if (retval < 0) {
-      cvProcessError(cv_mem, -1, "CVBBDPRE",
-                     "CVBBDPrecSetup", MSGBBD_SUNMAT_FAIL);
-      return(-1);
-    }
-    if (retval > 0) {
-      return(1);
+    SUNCheckCallNoRet(retval);
+    if (retval) {
+      cvProcessError(cv_mem, CV_SUNMAT_FAIL, __LINE__, __func__, 
+                     __FILE__, MSGBBD_SUNMAT_FAIL);
+      return(SUNLS_UNRECOV_FAILURE);
     }
 
   /* Otherwise call CVBBDDQJac for new J value */
   } else {
 
     *jcurPtr = SUNTRUE;
     retval = SUNMatZero(pdata->savedJ);
-    if (retval < 0) {
-      cvProcessError(cv_mem, -1, "CVBBDPRE",
-                     "CVBBDPrecSetup", MSGBBD_SUNMAT_FAIL);
-      return(-1);
-    }
-    if (retval > 0) {
-      return(1);
+    SUNCheckCallNoRet(retval);
+    if (retval) {
+      cvProcessError(cv_mem, CV_SUNMAT_FAIL, __LINE__, __func__, 
+                     __FILE__, MSGBBD_SUNMAT_FAIL);
+      return(SUNLS_UNRECOV_FAILURE);
     }
 
     retval = CVBBDDQJac(pdata, t, y, pdata->tmp1,
                         pdata->tmp2, pdata->tmp3);
     if (retval < 0) {
```

Comment:
Should this have a call to the new error-checking, e.g.
```suggestion
    SUNCheckCallNoRet(retval);
    if (retval < 0) {
```